### DISCUSSION TITLE
[Feature request] Syncing sidebar websites, UI customizations and keyboard shortcuts.

### DISCUSSION THREAD

**Discussion by Axenide** [zen-browser/desktop#1945] [Ideas]
Hi, I think having the ability to sync those things would be great (if possible, of course).

Right now I extract some lines from `prefs.js` to my user.js to keep these things between devices (using my dotfiles), but of course this has some drawbacks like having to search for the changes in `prefs.js` and copy them to `user.js` every time I want to modify anything.
About keyboard shortcuts, I have `zen-keyboard-shortcuts.js` in my dotfiles too, and it works most of the time (because sometimes it gets overriden by the defaults for some reason).

Anyway, this could be done for lots of things like configs and even the userChrome.css file, though I am completely ignorant of what sync can and cannot do.

**Comment by harrrrshit** [comment]
## Agreed!

- **Since Zen offers a lot of features, it's not quick to setup everything the way you need/like/want.** 
_But what if we can have all the settings, extensions, zen mods, bookmarks, sidebar width size, pinned tabs, opened tabs, workspaces, basically everything into a file that we can export import as per our need. This would speedup the process of setting up the browser from 30-45mins to just 5-10mins. Just setup extension settings, login to sites, and you are probably done._
 
- **In this case Firefox Sync isn't a great alternative since it doesn't even sync/save all the browser in-built settings like dns settings and much more.**
_But if it's possible to get everything A-Z synced through Firefox sync that too works, although the traditional file export and import option do have an upper hand since it doesn't require the users to create a Firefox account if they don't have one.
For Example: How UBlock Origin and many other extension keeps all their preferences and user data into a file that user can import export which can bring the user where they left off or just keep the file as an backup._

- **Zen browser being in heavy development and in it's alpha stage. It would also help in situation Zen Browser crashes for unknown reason and all the user tabs goes away.** 
_Many user have a ton of tabs opened on their browser, and this feature could provide relief from one of their biggest fear of losing all their work._ 
## Maybe through an update or some crash, and all of a sudden, all their tabs are gone.

**Reply by harrrrshit** [reply]
@mauro-balades Would love to know your opinion about the same.
